src/explore-education-statistics-common/src/modules/table-tool/components/TableHeadersForm.tsx

```diff
@@ -50,113 +56,202 @@ const TableHeadersForm = ({
+                          destinationId,
+                          reorder(
+                            form.values[destinationId] as unknown[],
+                            source.index,
+                            destination.index,
+                          ),
+                        );
+
+                        return;
+                      }
+
+                      // Moving group to the other axis
+                      const nextSourceValue = [...form.values[sourceId]];
+                      const nextDestinationValue = [
+                        ...form.values[destinationId],
+                      ];
+                      const [sourceItem] = nextSourceValue.splice(
                         source.index,
+                        1,
+                      );
+                      nextDestinationValue.splice(
                         destination.index,
-                      ),
-                    );
-
-                    return;
-                  }
-
-                  const nextSourceValue = [...form.values[sourceId]];
-                  const nextDestinationValue = [...form.values[destinationId]];
-
-                  const [sourceItem] = nextSourceValue.splice(source.index, 1);
-                  nextDestinationValue.splice(destination.index, 0, sourceItem);
-
-                  form.setFieldValue(sourceId, nextSourceValue);
-                  form.setFieldValue(destinationId, nextDestinationValue);
-
-                  form.setFieldTouched(sourceId);
-                  form.setFieldTouched(destinationId);
-                }}
-              >
-                <FormGroup>
-                  <div className="govuk-!-margin-bottom-2">
-                    <FormFieldSortableListGroup<
-                      PickByType<TableHeadersConfig, Filter[][]>
-                    >
-                      name="rowGroups"
-                      legend="Row groups"
-                      groupLegend="Row group"
-                    />
-                  </div>
-
-                  <div className="govuk-!-margin-bottom-2">
-                    <FormFieldSortableListGroup<
-                      PickByType<TableHeadersConfig, Filter[][]>
-                    >
-                      name="columnGroups"
-                      legend="Column groups"
-                      groupLegend="Column group"
-                    />
-                  </div>
-                </FormGroup>
-              </DragDropContext>
-
-              <Button type="submit">Re-order table</Button>
-            </Form>
-          );
-        }}
-      </Formik>
-    </Details>
+                        0,
+                        sourceItem,
+                      );
+
+                      form.setFieldValue(sourceId, nextSourceValue);
+                      form.setFieldValue(destinationId, nextDestinationValue);
+                      form.setFieldTouched(sourceId);
+                      form.setFieldTouched(destinationId);
```

Comment:
We might want to just extract this out into a `handleDragEnd` callback as it's a bit unwieldy currently.